runtime/src/bank/fee_distribution.rs

```diff
@@ -0,0 +1,489 @@
+use {
+    super::Bank,
+    log::{debug, error, warn},
+    solana_accounts_db::{account_rent_state::RentState, stake_rewards::RewardInfo},
+    solana_sdk::{
+        account::{ReadableAccount, WritableAccount},
+        pubkey::Pubkey,
+        reward_type::RewardType,
+    },
+    solana_vote::vote_account::VoteAccountsHashMap,
+    std::sync::atomic::Ordering::Relaxed,
+};
+
+impl Bank {
+    // Distribute collected transaction fees for this slot to collector_id (= current leader).
+    //
+    // Each validator is incentivized to process more transactions to earn more transaction fees.
+    // Transaction fees are rewarded for the computing resource utilization cost, directly
+    // proportional to their actual processing power.
+    //
+    // collector_id is rotated according to stake-weighted leader schedule. So the opportunity of
+    // earning transaction fees are fairly distributed by stake. And missing the opportunity
+    // (not producing a block as a leader) earns nothing. So, being online is incentivized as a
+    // form of transaction fees as well.
+    //
+    // On the other hand, rent fees are distributed under slightly different philosophy, while
+    // still being stake-weighted.
+    // Ref: distribute_rent_to_validators
+    pub(super) fn distribute_transaction_fees(&self) {
```

Comment:
In future, please limit move commits to just the move by itself and rename methods in a separate commit.